meilisearch-http/src/routes/dump.rs

```diff
@@ -92,16 +96,16 @@ impl tokio::stream::Stream for StreamBody {
     ) -> Poll<Option<Self::Item>> {
         use std::ops::DerefMut;
         let stream = self.deref_mut();
-            match stream.file.read(&mut stream.buffer).await {
-                Ok(count) => {
-                    if count > 0 {
-                        Poll::Ready(Some(Ok(Bytes::copy_from_slice(&self.buffer[..count]))))
-                    } else {
-                        Poll::Ready(None)
-                    }
-                },
-                Err(e) => Poll::Ready(Some(Err(Error::dump_read_failed(e).into())))
-            }
+        match stream.file.read(&mut stream.buffer).await {
+            Ok(count) => {
+                if count > 0 {
+                    Poll::Ready(Some(Ok(Bytes::copy_from_slice(&self.buffer[..count]))))
+                } else {
+                    Poll::Ready(None)
+                }
+            },
+            Err(e) => Poll::Ready(Some(Err(Error::dump_read_failed(e).into())))
+        }
```

Comment:
that's not right indeed. You are trying to use await out of an async context. This is a lower level of abstraction, and we are getting into the internal functioning of futures in rust. I would suggest your read https://cfsamson.github.io/books-futures-explained/ in order to get a better understanding of how it works.

In the mean time, what you want to use is [this](https://docs.rs/tokio/0.2.22/tokio/io/trait.AsyncRead.html#tymethod.poll_read)

Reply:
I haven't tested it, but something like this could work:
```rust
impl tokio::stream::Stream for StreamBody {
    type Item = Result<Bytes, ResponseError>;
    fn poll_next(
        mut self: Pin<&mut Self>,
        cx: &mut Context<'_>,
    ) -> Poll<Option<Self::Item>> {
        //  you can't hold to mut refs to self, need to take ownership on the buffer for some time
        let mut buffer = std::mem::replace(&mut self.buffer, vec![]);
        let file = &mut self.file;
        pin_utils::pin_mut!(file);
        match file.poll_read(cx, &mut buffer) {
            Poll::Ready(Ok(size)) => {
                if size > 0 {
                    // place it back when done with it
                    let _ = std::mem::replace(&mut self.buffer, buffer);
                    Poll::Ready(Some(Ok(Bytes::copy_from_slice(&self.buffer[..size]))))
                } else {
                    Poll::Ready(None)
                }
            }
            Poll::Ready(Err(e)) => Poll::Ready(Some(Err(Error::Internal(e.to_string()).into()))),
            Poll::Pending => Poll::Pending,
        }
    }
}
```

Reply:
But, just for sake of clarity, this is a feature we have thought about, but we are still not sure we want to have it, there are many ongoing discussion about that. I'm happy to give you mentorship on this, and get you going, but keep in mind that nothing is decided about the future of this feature :)

Reply:
@MarinPostma I had to change your code a bit, it was not working because `vec![]` was causing the second poll to return an empty vec. I also replaced  `pin_utils::pin_mut!(file);` with `Pin::new(&mut self.file);` as suggested by @Koxiaet